File: programming_languages/Python/code/T32/test_T32.py
import os
import Test1.test as file1
import Test2.test as file2

files_to_compare = [
    ("Test1/test.py", "Test1_Nuitka/test.py"),
    ("Test2/test.py", "Test2_Nuitka/test.py")
]

def test_files_identical():
    for orig, nuitka in files_to_compare:
        assert os.path.exists(orig), f"Original file not found: {orig}"
        assert os.path.exists(nuitka), f"Nuitka file not found: {nuitka}"

        with open(orig, "r", encoding="utf-8") as f1:
            content1 = f1.read()
        with open(nuitka, "r", encoding="utf-8") as f2:
            content2 = f2.read()

        assert content1 == content2, f"Files differ: {orig} != {nuitka}"

def test_identical_output_check_all():
    f, e, d, c = 4, 3, 2, 1
    lookup = [f, e, d, d, c, c, c, c]

    combinations = [(b1, b2, b3) for b1 in [0, 1] for b2 in [0, 1] for b3 in [0, 1]]
    
    for b1, b2, b3 in combinations:
        r1 = file1.test(b1, b2, b3, c, d, e, f, None)
        r2 = file2.test(b1, b2, b3, c, d, e, f, lookup)

        for name, r in [("file1", r1), ("file2", r2)]:
            assert isinstance(r, int), f"{name}.test should return a number, got {type(r)}"

        if not (r1 == r2):
            raise AssertionError(f"Mismatch detected between versions!")

if __name__ == "__main__":
    test_files_identical()
    test_identical_output_check_all()

File: programming_languages/Java/code/T32/test_T32.java
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;

public class test_T32 {

    public static void main(String[] args) {
        test_files_identical();
        test_identical_output_check_all();
    }

    public static void test_files_identical() {
        String[][] filesToCompare = {
            {"Test1/test.java", "Test1_Xint/test.java"},
            {"Test2/test.java", "Test2_Xint/test.java"}
        };

        for (String[] pair : filesToCompare) {
            try {
                Path pathOrig = Paths.get(pair[0]);
                Path pathXint = Paths.get(pair[1]);

                if (!Files.exists(pathOrig) || !Files.exists(pathXint)) {
                    System.err.println("[ERROR] File not found: " + pair[0] + " or " + pair[1]);
                    System.exit(1);
                }

                java.util.List<String> f1Lines = Files.readAllLines(pathOrig);
                java.util.List<String> f2Lines = Files.readAllLines(pathXint);

                f1Lines.removeIf(line -> ignoreLine(line));
                f2Lines.removeIf(line -> ignoreLine(line));

                if (!f1Lines.equals(f2Lines)) {
                    System.err.println("[ERROR] Files differ (ignoring package & REPS): "
                            + pair[0] + " != " + pair[1]);
                    System.exit(1);
                }

            } catch (IOException e) {
                System.err.println("[ERROR] Error reading files: " + e.getMessage());
                System.exit(1);
            }
        }
    }

    private static boolean ignoreLine(String line) {
        String l = line.trim();

        return l.startsWith("package ")
            || l.startsWith("public static final int REPS")
            || l.startsWith("public static final long REPS")
            || l.contains("for (int reps")
            || l.contains("for (long reps");
    }

    public static void test_identical_output_check_all() {
        try {
            int f = 4, e = 3, d = 2, c = 1;
            int[] lookup = new int[]{f, e, d, d, c, c, c, c};

            int[] values = {0, 1};

            for (int b1 : values) {
                for (int b2 : values) {
                    for (int b3 : values) {

                        int r1 = Test1.test.test(b1, b2, b3, c, d, e, f, null);
                        int r2 = Test2.test.test(b1, b2, b3, c, d, e, f, lookup);

                        if (!(r1 == r2)) {
                            System.err.println("[ERROR] Mismatch detected between versions!");
                            System.exit(1);
                        }
                    }
                }
            }

            System.exit(0);

        } catch (Throwable e) {
            System.err.println("[ERROR] Execution error: " + e.getMessage());
            System.exit(1);
        }
    }
}